### PR TITLE
Fix the error caused by the same path and alias of RouteConfig

### PR DESCRIPTION
**Pull request by chenfengjw163**

### CODE REVIEW

**Comment by chenfengjw163** [comment]
#2461

File: src/create-route-map.js [hunk 1]
```diff
@@ -112,32 +112,40 @@ function addRouteRecord (
...
-    })
-  }
-
   if (!pathMap[record.path]) {
     pathList.push(record.path)
     pathMap[record.path] = record
   }
 
+  if (route.alias !== undefined) {
+    if (route.alias === path) {
```
Comment (posva) [inline src/create-route-map.js]: The warn should only appear on dev mode. Also could you add a test for it?

**Review by posva — CHANGES_REQUESTED** [review]

File: src/create-route-map.js (hunk 1 above)
Comment (chenfengjw163) [reply src/create-route-map.js]: OK, I have already added

**Comment by posva** [comment]
Hey, I added a check for arrays too and directly merged to `dev`. Thank you!

**Pull request closed without merging**